apps/req.c

```diff
@@ -1153,56 +1158,22 @@ static int add_DN_object(X509_NAME *n, char *text, const char *def,
                          char *value, int nid, int n_min, int n_max,
                          unsigned long chtype, int mval)
 {
-    int i, ret = 0;
+    int ret = 0;
     char buf[1024];
- start:
-    if (!batch)
-        BIO_printf(bio_err, "%s [%s]:", text, def);
-    (void)BIO_flush(bio_err);
-    if (value != NULL) {
-        if (!join(buf, sizeof(buf), value, "\n", "DN value"))
-            return 0;
-        BIO_printf(bio_err, "%s\n", value);
-    } else {
-        buf[0] = '\0';
-        if (!batch) {
-            if (!fgets(buf, sizeof buf, stdin))
-                return 0;
-        } else {
-            buf[0] = '\n';
-            buf[1] = '\0';
-        }
-    }
-
-    if (buf[0] == '\0')
-        return 0;
-    if (buf[0] == '\n') {
-        if ((def == NULL) || (def[0] == '\0'))
-            return 1;
-        if (!join(buf, sizeof(buf), def, "\n", "DN default"))
-            return 0;
-    } else if ((buf[0] == '.') && (buf[1] == '\n')) {
-        return 1;
-    }
-
-    i = strlen(buf);
-    if (buf[i - 1] != '\n') {
-        BIO_printf(bio_err, "weird input :-(\n");
-        return 0;
-    }
-    buf[--i] = '\0';
-#ifdef CHARSET_EBCDIC
-    ebcdic2ascii(buf, buf, i);
-#endif
-    if (!req_check_len(i, n_min, n_max)) {
-        if (batch || value)
-            return 0;
-        goto start;
-    }
+    const char *desc1 = "DN value";
```

Comment:
Why aren't these inline in the function call?  There is no need to have local variables for the descriptions.

Reply:
@paulidale Thank you for review.

These variables of the description are used in join() when buffer size is small.
Error message are 4 patterns.
- add_attribute_object has 2 patterns.
- add_DN_object has 2 patterns.

Reply:
I don't think I worded it well.  I didn't mean to inline them into build_data itself, just in the call to build_data:
```
ret = build_data(text, def, value, n_min, n_max, buf, sizeof(buf), "DN value", "DN default");
```

and similarly for the other case.

They are only used once as an argument for the build_data call, this doesn't warrant a local variable IMO.

Reply:
@paulidale Thank you for the explanation. I think so, too. And fix it.